test/controllers/FunctionalTest.java

```diff
@@ -34,11 +36,10 @@ public void testRejectWebSocket() {
                 try {
                     String serverURL = "ws://localhost:37117/ws";
                     WebSocketClient.LoggingListener listener = new WebSocketClient.LoggingListener(message -> {});
-                    CompletableFuture<WebSocket> completionStage = webSocketClient.call(serverURL, listener);
+                    CompletableFuture<NettyWebSocket> completionStage = webSocketClient.call(serverURL, serverURL, listener);
                     await().until(completionStage::isDone);
-                    assertThat(completionStage)
-                            .hasFailedWithThrowableThat()
-                            .hasMessageContaining("Invalid Status Code 403");
+                    assertThat(completionStage.get())
+                        .isNull();
```

Comment:
This assertion is wrong. Something changed in AHC that altered the behavior so that instead of getting a failed `CompletionStage` with a `Invalid Status Code 403` message, the result is not a `completed` `CompletionStage` that contains a `null`.
The underlying version fo AHC had a rather large [WebSocket revamp](https://github.com/AsyncHttpClient/async-http-client/commit/98bef40637212fc1c64ec47b1b11bb1a15144704)  which probably introduced this change.

Reply:
@renatocaval did you have the chance to review this comment/change?